flink-table/flink-table-common/src/main/java/org/apache/flink/table/api/dataview/ListView.java

```diff
@@ -31,13 +31,16 @@
 import java.util.Objects;
 
 /**
- * A {@link DataView} that provides {@link List}-like functionality in the accumulator of an {@link
- * AggregateFunction} or {@link TableAggregateFunction} when large amounts of data are expected.
+ * A {@link DataView} that provides {@link List}-like functionality in state entries.
  *
  * <p>A {@link ListView} can be backed by a Java {@link ArrayList} or can leverage Flink's state
- * backends depending on the context in which the aggregate function is used. In many unbounded data
- * scenarios, the {@link ListView} delegates all calls to a {@link ListState} instead of the {@link
- * ArrayList}.
+ * backends depending on the context. In many unbounded data scenarios, the {@link ListView}
```

Comment:
I am wondering if it would be helpful to be more precise, and list the cases when it delegates calls to the List state and when ArrayList is used and the reasoning.

Reply:
I improved the docs, but aggregate functions are highly complex and used at various locations. If we really want to list all special cases, this would require a dedicated task; out of scope for this PR.